pkg/adaptor/cloud/vsphere/provider.go

```diff
@@ -49,6 +54,27 @@ func NewProvider(config *Config) (cloud.Provider, error) {
 	return provider, nil
 }
 
+func checkConfig(config *Config) error {
+
+	// Do some initial checks of the optional input values
+
+	if config.DRS == "true" {
+		if config.Cluster == "" {
+			return fmt.Errorf("Error: A cluster name is required with DRS")
+		}
```

Comment:
nit: i wonder if it make sense to have fallback try to use Host and Datastore in cases where DRS=true but cluster is not set

Reply:
The problem with that is the user has misconfigured the CAA and we need to point that out. They are asking for DRS and if we are not doing what they ask to try to help them it may sound good but it also leads to confusion. Meaning DRS requires a cluster if they have omitted a cluster name they have ignored the most crucial part of the config.